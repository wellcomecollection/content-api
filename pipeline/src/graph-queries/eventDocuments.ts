const query = `{
  events {
    title
    isOnline
    availableOnline
    format {
      ... on event-formats {
        title
      }
    }
    contributors {
      role {
        title
      }
      contributor {
        ... on people {
          name
        }
        ... on organisations {
          name
        }
      }
    }
    interpretations {
      interpretationType {
        title
      }
    }
    times {
      startDateTime
      endDateTime
      isFullyBooked
      onlineIsFullyBooked
    }
    audiences {
      audience {
        title
      }
    }
    schedule {
      event {
        title
        isOnline
        format {
          ... on event-formats {
            title
          }
        }
        interpretations {
          interpretationType {
            title
          }
        }
        audiences {
          audience {
            title
          }
        }
        locations {
          location {
            title
          }
        }
        times {
          startDateTime
          endDateTime
        }
      }
      isNotLinked
    }
    locations {
      location {
        title
      }
    }
    promo {
      ... on editorialImage {
        non-repeat {
          caption
          image
        }
      }
    }
    series {
      series {
        title
        contributors {
          contributor {
            ... on people {
              name
            }
            ... on organisations {
              name
            }
          }
        }
      }
    }
  }
}`;

export default query;
