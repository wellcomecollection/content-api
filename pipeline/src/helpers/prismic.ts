import * as prismic from "@prismicio/client";
import { ContentType } from "../types";

const graphQueryArticles = `{
    articles {
      title
      format {
        title
      }
      promo
      contributors {
        ...contributorsFields
        role {
          title
        }
        contributor {
          ... on people {
            name
          }
          ... on organisations {
            name
          }
        }
      }
    }
  }`.replace(/\n(\s+)/g, "\n"); // Pre-emptive removal of whitespaces as requests to the Prismic Rest API are limited to 2048 characters.

// TODO Review what prismic function we use here.
// getAllByType (no pagination blocker but only one type at a time)
// or getByType (which allows more than one type at a time).
// Review if predicates are still required
type GetPrismicDocumentsParams = {
  client: prismic.Client;
  contentTypes: ContentType[];
  after?: string;
};

export type PrismicPage<T> = {
  docs: T[];
  lastDocId?: string;
};

export const getPrismicDocuments = async <T>({
  client,
  contentTypes,
  after,
}: GetPrismicDocumentsParams): Promise<PrismicPage<T>> => {
  const docs = await client.getByType(contentTypes[0], {
    graphQuery: graphQueryArticles,
    predicates: [prismic.predicate.any("document.type", contentTypes)],
    orderings: {
      field: "document.last_publication_date",
      direction: "desc",
    },
    pageSize: 100,
    after,
  });

  const results = docs.results;
  const lastDoc = results[results.length - 1];
  const lastDocId = lastDoc?.id;

  return {
    docs: results as T[],
    lastDocId,
  };
};
