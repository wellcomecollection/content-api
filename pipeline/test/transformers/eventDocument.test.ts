import { transformEventDocument } from "@weco/content-pipeline/src/transformers/eventDocument";
import { EventPrismicDocument } from "@weco/content-pipeline/src/types/prismic/eventDocuments";
import { forEachPrismicSnapshot } from "@weco/content-pipeline/test/fixtures/prismic-snapshots";

describe("eventDocument transformer", () => {
  forEachPrismicSnapshot<EventPrismicDocument>("events")(
    "transforms events from Prismic to the expected format",
    (prismicDocument) => {
      const transformed = transformEventDocument(prismicDocument);
      // Unsure why it errors, it _is_ in a test() function, see `forEachPrismicSnapshot`
      // eslint-disable-next-line jest/no-standalone-expect
      expect(transformed).toMatchSnapshot();
    },
  );
});
