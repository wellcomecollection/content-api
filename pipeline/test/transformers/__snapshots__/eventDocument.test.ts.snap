// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eventDocument transformer transforms articles from Prismic to the expected format (document: events/WqfhAyUAAM8VVngg) 1`] = `
{
  "display": {
    "format": {
      "id": "Wn3Q3SoAACsAIeFI",
      "label": "Performance",
      "type": "EventFormat",
    },
    "id": "WqfhAyUAAM8VVngg",
    "image": undefined,
    "interpretations": [],
    "locations": [],
    "times": [],
    "title": "DMC Test",
    "type": "Event",
  },
  "id": "WqfhAyUAAM8VVngg",
  "query": {
    "caption": undefined,
    "linkedIdentifiers": [
      "Wn3Q3SoAACsAIeFI",
    ],
    "series": [],
    "title": "DMC Test",
  },
}
`;

exports[`eventDocument transformer transforms articles from Prismic to the expected format (document: events/ZFt0WhQAAHnPEH7P) 1`] = `
{
  "display": {
    "format": {
      "id": "W5fV5iYAACQAMxHb",
      "label": "Festival",
      "type": "EventFormat",
    },
    "id": "ZFt0WhQAAHnPEH7P",
    "image": {
      "16:9": {
        "alt": "Photograph of a farmer getting ready to harvest his organic crop of native Ragi in Kariyappanadoddi, India. Behind him are hills and a small single storey building. Interwoven into the scene is a graphic element made up of thin red drawn circular lines, which create a larger organic pattern behind the farmer.",
        "copyright": "Farmer Basavaraj in Kariyappanadoddi inspects his ragi crop before harvesting near Bannerghatta, India
 | | | | | Quicksand |",
        "dimensions": {
          "height": 1800,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": 0,
          "y": 0,
          "zoom": 0.8,
        },
        "id": "ZGYrjxQAADxsO9nG",
        "url": "https://images.prismic.io/wellcomecollection/0793d963-8807-4525-a3d7-e77fa4435baa_LBE+Key+Image+04.jpg?auto=compress,format&rect=0,0,4000,2250&w=3200&h=1800",
      },
      "32:15": {
        "alt": "Photograph of a farmer getting ready to harvest his organic crop of native Ragi in Kariyappanadoddi, India. Behind him are hills and a small single storey building. Interwoven into the scene is a graphic element made up of thin red drawn circular lines, which create a larger organic pattern behind the farmer.",
        "copyright": "Farmer Basavaraj in Kariyappanadoddi inspects his ragi crop before harvesting near Bannerghatta, India
 | | | | | Quicksand |",
        "dimensions": {
          "height": 1500,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": 0,
          "y": -185,
          "zoom": 0.8,
        },
        "id": "ZGYrjxQAADxsO9nG",
        "url": "https://images.prismic.io/wellcomecollection/0793d963-8807-4525-a3d7-e77fa4435baa_LBE+Key+Image+04.jpg?auto=compress,format&rect=0,232,4000,1875&w=3200&h=1500",
      },
      "alt": "Photograph of a farmer getting ready to harvest his organic crop of native Ragi in Kariyappanadoddi, India. Behind him are hills and a small single storey building. Interwoven into the scene is a graphic element made up of thin red drawn circular lines, which create a larger organic pattern behind the farmer.",
      "copyright": "Farmer Basavaraj in Kariyappanadoddi inspects his ragi crop before harvesting near Bannerghatta, India
 | | | | | Quicksand |",
      "dimensions": {
        "height": 2250,
        "width": 4000,
      },
      "edit": {
        "background": "#fff",
        "x": 0,
        "y": 0,
        "zoom": 1,
      },
      "id": "ZGYrjxQAADxsO9nG",
      "square": {
        "alt": "Photograph of a farmer getting ready to harvest his organic crop of native Ragi in Kariyappanadoddi, India. Behind him are hills and a small single storey building. Interwoven into the scene is a graphic element made up of thin red drawn circular lines, which create a larger organic pattern behind the farmer.",
        "copyright": "Farmer Basavaraj in Kariyappanadoddi inspects his ragi crop before harvesting near Bannerghatta, India
 | | | | | Quicksand |",
        "dimensions": {
          "height": 3200,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": -1709,
          "y": 0,
          "zoom": 1.4222222222222223,
        },
        "id": "ZGYrjxQAADxsO9nG",
        "url": "https://images.prismic.io/wellcomecollection/0793d963-8807-4525-a3d7-e77fa4435baa_LBE+Key+Image+04.jpg?auto=compress,format&rect=1202,0,2250,2250&w=3200&h=3200",
      },
      "type": "PrismicImage",
      "url": "https://images.prismic.io/wellcomecollection/0793d963-8807-4525-a3d7-e77fa4435baa_LBE+Key+Image+04.jpg?auto=compress,format",
    },
    "interpretations": [],
    "locations": [
      {
        "id": "XEnJIhUAAArdgVsW",
        "label": "Throughout our building",
        "type": "EventLocation",
      },
      {
        "id": "ef04c8e3-26be-4fbc-9bef-f52589ebc56c",
        "label": "Online",
        "type": "EventLocation",
      },
    ],
    "times": [
      {
        "endDateTime": 2023-06-23T17:00:00.000Z,
        "startDateTime": 2023-06-23T09:00:00.000Z,
      },
    ],
    "title": "Land Body Ecologies Festival Day Two",
    "type": "Event",
  },
  "id": "ZFt0WhQAAHnPEH7P",
  "query": {
    "caption": "Explore the deep connections between mental health and ecosystem health in this four-day festival created by Land Body Ecologies (LBE), from teams across India, Uganda, Kenya, Thailand, the UK and the Arctic. Take part and experience performances, food, workshops, music, films and more.",
    "linkedIdentifiers": [
      "W5fV5iYAACQAMxHb",
      "XEnJIhUAAArdgVsW",
      "ZFtubBQAAHnPEGEG",
      "ZFteSBQAADxsEBMa",
      "ZFtfvxQAAHnPEBo-",
      "ZFtX3xQAADxsD_RB",
      "ZFtd-BQAAMCZEBGP",
      "ZFtZFBQAAHnPD_n1",
      "ZGTv2hQAADxsOdmn",
      "ZFttJBQAAHnPEFn0",
      "ZFtr3BQAAHnPEFN5",
      "ZFttnRQAAHnPEFyu",
      "ZFtsuRQAAHnPEFfW",
      "ZFtsVxQAAHnPEFXh",
      "ZFuh7hQAAMCZEVTX",
      "XwRXehQAAOMZ-avh",
      "ZGtEqBQAACi-Q2xc",
      "ZIbqVRAAAMwEHpqg",
      "ZFt3UBQAAMFmEIya",
    ],
    "series": [
      {
        "contributors": [
          "Land Body Ecologies",
          "The Hub",
        ],
        "id": "ZFt3UBQAAMFmEIya",
        "title": "Land Body Ecologies Festival",
      },
    ],
    "title": "Land Body Ecologies Festival Day Two",
  },
}
`;

exports[`eventDocument transformer transforms articles from Prismic to the expected format (document: events/ZJLZoRAAACIARz42) 1`] = `
{
  "display": {
    "format": {
      "id": "WmYRpCQAACUAn-Ap",
      "label": "Gallery tour",
      "type": "EventFormat",
    },
    "id": "ZJLZoRAAACIARz42",
    "image": {
      "16:9": {
        "alt": "Photograph of an art installation in gallery space. It's a dark space with black glossy floors and walls. A large parachute suspended from the ceiling which is lit with vibrant pink and orange light. Two visitors are looking at the instillation.",
        "copyright": "From the Collection: For What It’s Worth, 2023. Jess Dobkin |  Gallery photo: Steven Pocock | | | CC-BY-NC | |",
        "dimensions": {
          "height": 1800,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": 0,
          "y": 0,
          "zoom": 0.8,
        },
        "id": "ZKgZeBAAACEAfpCr",
        "url": "https://images.prismic.io/wellcomecollection/3e1ef5ca-cf7f-40e5-b1dc-0730a5e11225_EP_002225_007-full.jpg?auto=compress,format&rect=0,0,4000,2250&w=3200&h=1800",
      },
      "32:15": {
        "alt": "Photograph of an art installation in gallery space. It's a dark space with black glossy floors and walls. A large parachute suspended from the ceiling which is lit with vibrant pink and orange light. Two visitors are looking at the instillation.",
        "copyright": "From the Collection: For What It’s Worth, 2023. Jess Dobkin |  Gallery photo: Steven Pocock | | | CC-BY-NC | |",
        "dimensions": {
          "height": 1500,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": 0,
          "y": -150,
          "zoom": 0.8,
        },
        "id": "ZKgZeBAAACEAfpCr",
        "url": "https://images.prismic.io/wellcomecollection/3e1ef5ca-cf7f-40e5-b1dc-0730a5e11225_EP_002225_007-full.jpg?auto=compress,format&rect=0,188,4000,1875&w=3200&h=1500",
      },
      "alt": "Photograph of an art installation in gallery space. It's a dark space with black glossy floors and walls. A large parachute suspended from the ceiling which is lit with vibrant pink and orange light. Two visitors are looking at the instillation.",
      "copyright": "From the Collection: For What It’s Worth, 2023. Jess Dobkin |  Gallery photo: Steven Pocock | | | CC-BY-NC | |",
      "dimensions": {
        "height": 2250,
        "width": 4000,
      },
      "edit": {
        "background": "#fff",
        "x": 0,
        "y": 0,
        "zoom": 1,
      },
      "id": "ZKgZeBAAACEAfpCr",
      "square": {
        "alt": "Photograph of an art installation in gallery space. It's a dark space with black glossy floors and walls. A large parachute suspended from the ceiling which is lit with vibrant pink and orange light. Two visitors are looking at the instillation.",
        "copyright": "From the Collection: For What It’s Worth, 2023. Jess Dobkin |  Gallery photo: Steven Pocock | | | CC-BY-NC | |",
        "dimensions": {
          "height": 3200,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": -1935,
          "y": 0,
          "zoom": 1.4222222222222223,
        },
        "id": "ZKgZeBAAACEAfpCr",
        "url": "https://images.prismic.io/wellcomecollection/3e1ef5ca-cf7f-40e5-b1dc-0730a5e11225_EP_002225_007-full.jpg?auto=compress,format&rect=1361,0,2250,2250&w=3200&h=3200",
      },
      "type": "PrismicImage",
      "url": "https://images.prismic.io/wellcomecollection/3e1ef5ca-cf7f-40e5-b1dc-0730a5e11225_EP_002225_007-full.jpg?auto=compress,format",
    },
    "interpretations": [
      {
        "id": "XkFGqxEAACIAIhNH",
        "label": "British Sign Language",
        "type": "EventInterpretation",
      },
    ],
    "locations": [
      {
        "id": "W22bICkAACgA-hVJ",
        "label": "Information Point",
        "type": "EventLocation",
      },
    ],
    "times": [
      {
        "endDateTime": 2023-08-24T17:30:00.000Z,
        "startDateTime": 2023-08-24T16:30:00.000Z,
      },
    ],
    "title": "Perspective Tour With Jess Dobkin",
    "type": "Event",
  },
  "id": "ZJLZoRAAACIARz42",
  "query": {
    "caption": "Join performance artist Jess Dobkin for a tour of our exhibition ‘Milk’.",
    "linkedIdentifiers": [
      "WmYRpCQAACUAn-Ap",
      "W22bICkAACgA-hVJ",
      "XkFGqxEAACIAIhNH",
      "Wyu-wCoAACkACpod",
      "ZLeRyBAAACIAwnNz",
      "WlYUDCQAACQAWcdt",
    ],
    "series": [
      {
        "contributors": [],
        "id": "WlYUDCQAACQAWcdt",
        "title": "Perspective Tours",
      },
    ],
    "title": "Perspective Tour With Jess Dobkin",
  },
}
`;

exports[`eventDocument transformer transforms articles from Prismic to the expected format (document: events/ZQgdkREAAAbr6cRR) 1`] = `
{
  "display": {
    "format": {
      "id": "dfc2b7f9-c362-47da-9644-b0f98212ccaa",
      "label": "Event",
      "type": "EventFormat",
    },
    "id": "ZQgdkREAAAbr6cRR",
    "image": {
      "16:9": {
        "alt": "A Visitor Experience Assistant standing in a brightly lit gallery giving a tour of the exhibition to visitors. Two museum staff are wearing black t-shirts with "Wellcome Collection" printed in white lettering on their sleeves. ",
        "copyright": "Audio-described tour of 'The Cult of Beauty' exhibition | Gallery photo: Steven Pocock | Wellcome Collection | | CC-BY-NC | |",
        "dimensions": {
          "height": 1800,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": 0,
          "y": 0,
          "zoom": 0.8,
        },
        "id": "ZJ2kURAAACIAdBpu",
        "url": "https://images.prismic.io/wellcomecollection/d8e4d673-0874-4888-88f6-ff3ebd7de642_EP_002351_013+4k+16x9.jpg?auto=compress,format&rect=0,0,4000,2250&w=3200&h=1800",
      },
      "32:15": {
        "alt": "A Visitor Experience Assistant standing in a brightly lit gallery giving a tour of the exhibition to visitors. Two museum staff are wearing black t-shirts with "Wellcome Collection" printed in white lettering on their sleeves. ",
        "copyright": "Audio-described tour of 'The Cult of Beauty' exhibition | Gallery photo: Steven Pocock | Wellcome Collection | | CC-BY-NC | |",
        "dimensions": {
          "height": 1500,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": 0,
          "y": -150,
          "zoom": 0.8,
        },
        "id": "ZJ2kURAAACIAdBpu",
        "url": "https://images.prismic.io/wellcomecollection/d8e4d673-0874-4888-88f6-ff3ebd7de642_EP_002351_013+4k+16x9.jpg?auto=compress,format&rect=0,188,4000,1875&w=3200&h=1500",
      },
      "alt": "A Visitor Experience Assistant standing in a brightly lit gallery giving a tour of the exhibition to visitors. Two museum staff are wearing black t-shirts with "Wellcome Collection" printed in white lettering on their sleeves. ",
      "copyright": "Audio-described tour of 'The Cult of Beauty' exhibition | Gallery photo: Steven Pocock | Wellcome Collection | | CC-BY-NC | |",
      "dimensions": {
        "height": 2250,
        "width": 4000,
      },
      "edit": {
        "background": "#fff",
        "x": 0,
        "y": 0,
        "zoom": 1,
      },
      "id": "ZJ2kURAAACIAdBpu",
      "square": {
        "alt": "A Visitor Experience Assistant standing in a brightly lit gallery giving a tour of the exhibition to visitors. Two museum staff are wearing black t-shirts with "Wellcome Collection" printed in white lettering on their sleeves. ",
        "copyright": "Audio-described tour of 'The Cult of Beauty' exhibition | Gallery photo: Steven Pocock | Wellcome Collection | | CC-BY-NC | |",
        "dimensions": {
          "height": 3200,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": -1244,
          "y": 0,
          "zoom": 1.4222222222222223,
        },
        "id": "ZJ2kURAAACIAdBpu",
        "url": "https://images.prismic.io/wellcomecollection/d8e4d673-0874-4888-88f6-ff3ebd7de642_EP_002351_013+4k+16x9.jpg?auto=compress,format&rect=875,0,2250,2250&w=3200&h=3200",
      },
      "type": "PrismicImage",
      "url": "https://images.prismic.io/wellcomecollection/d8e4d673-0874-4888-88f6-ff3ebd7de642_EP_002351_013+4k+16x9.jpg?auto=compress,format",
    },
    "interpretations": [],
    "locations": [
      {
        "id": "WrEgqSAAACAAPEw7",
        "label": "Gallery 1",
        "type": "EventLocation",
      },
    ],
    "times": [
      {
        "endDateTime": 2023-12-16T14:00:00.000Z,
        "startDateTime": 2023-11-23T16:00:00.000Z,
      },
    ],
    "title": "Lights Up on The Cult of Beauty",
    "type": "Event",
  },
  "id": "ZQgdkREAAAbr6cRR",
  "query": {
    "caption": "Explore ‘The Cult of Beauty’ exhibition with brighter and more even lighting conditions across the gallery. Audio and BSL digital guides, a visual story, and large-print and braille guides are available for use at these drop-in sessions.",
    "linkedIdentifiers": [
      "WrEgqSAAACAAPEw7",
      "ZQgi7hIAACQAN2QG",
      "ZQghdREAAEnw6cr1",
      "ZQgkIRIAACIAN2X1",
      "ZQgiFxYAACcAYG_6",
    ],
    "series": [],
    "title": "Lights Up on The Cult of Beauty",
  },
}
`;

exports[`eventDocument transformer transforms articles from Prismic to the expected format (document: events/ZRrijRIAAJNSARgG) 1`] = `
{
  "display": {
    "format": {
      "id": "Wn3Q3SoAACsAIeFI",
      "label": "Performance",
      "type": "EventFormat",
    },
    "id": "ZRrijRIAAJNSARgG",
    "image": {
      "16:9": {
        "alt": "A group of young performers on stage posing for a group portrait. ",
        "copyright": " | | | | | Photo by Sharne Fairchild |",
        "dimensions": {
          "height": 1800,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": 0,
          "y": 0,
          "zoom": 0.8,
        },
        "id": "ZVNa_hIAACcAarUk",
        "url": "https://images.prismic.io/wellcomecollection/a37c1539-a43a-490f-9f98-8c282d016a46_Group+shot+by+Sharne+Fairchild.jpg?auto=compress,format&rect=0,0,4000,2250&w=3200&h=1800",
      },
      "32:15": {
        "alt": "A group of young performers on stage posing for a group portrait. ",
        "copyright": " | | | | | Photo by Sharne Fairchild |",
        "dimensions": {
          "height": 1500,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": 0,
          "y": 0,
          "zoom": 0.8,
        },
        "id": "ZVNa_hIAACcAarUk",
        "url": "https://images.prismic.io/wellcomecollection/a37c1539-a43a-490f-9f98-8c282d016a46_Group+shot+by+Sharne+Fairchild.jpg?auto=compress,format&rect=0,0,4000,1875&w=3200&h=1500",
      },
      "alt": "A group of young performers on stage posing for a group portrait. ",
      "copyright": " | | | | | Photo by Sharne Fairchild |",
      "dimensions": {
        "height": 2250,
        "width": 4000,
      },
      "edit": {
        "background": "#fff",
        "x": 0,
        "y": 0,
        "zoom": 1,
      },
      "id": "ZVNa_hIAACcAarUk",
      "square": {
        "alt": "A group of young performers on stage posing for a group portrait. ",
        "copyright": " | | | | | Photo by Sharne Fairchild |",
        "dimensions": {
          "height": 3200,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": -1244.4444444444446,
          "y": 0,
          "zoom": 1.4222222222222223,
        },
        "id": "ZVNa_hIAACcAarUk",
        "url": "https://images.prismic.io/wellcomecollection/a37c1539-a43a-490f-9f98-8c282d016a46_Group+shot+by+Sharne+Fairchild.jpg?auto=compress,format&rect=875,0,2250,2250&w=3200&h=3200",
      },
      "type": "PrismicImage",
      "url": "https://images.prismic.io/wellcomecollection/a37c1539-a43a-490f-9f98-8c282d016a46_Group+shot+by+Sharne+Fairchild.jpg?auto=compress,format",
    },
    "interpretations": [
      {
        "id": "XkFGqxEAACIAIhNH",
        "label": "British Sign Language",
        "type": "EventInterpretation",
      },
      {
        "id": "W5JXVSYAACYAGtkh",
        "label": "Relaxed",
        "type": "EventInterpretation",
      },
    ],
    "locations": [
      {
        "id": "W8XfKxEAADwYer9D",
        "label": "The Studio and the Forum",
        "type": "EventLocation",
      },
    ],
    "times": [
      {
        "endDateTime": 2023-11-18T16:00:00.000Z,
        "startDateTime": 2023-11-18T14:00:00.000Z,
      },
    ],
    "title": "Standards, My Right to Beauty",
    "type": "Event",
  },
  "id": "ZRrijRIAAJNSARgG",
  "query": {
    "caption": "Join an afternoon of performance and discussion about beauty in which the audience plays an important part, including deciding on the running order and joining in debates.",
    "linkedIdentifiers": [
      "Wn3Q3SoAACsAIeFI",
      "W8XfKxEAADwYer9D",
      "XkFGqxEAACIAIhNH",
      "W5JXVSYAACYAGtkh",
      "WlYWECQAACcAWdBf",
      "WyvU0ioAACkACvwK",
      "ZRrmchIAACAAASlI",
      "ZRrnCxIAACAAASwU",
    ],
    "series": [],
    "title": "Standards, My Right to Beauty",
  },
}
`;

exports[`eventDocument transformer transforms articles from Prismic to the expected format (document: events/ZSPXJBAAACIAiERm) 1`] = `
{
  "display": {
    "format": {
      "id": "Wd-QYCcAACcAoiJS",
      "label": "Discussion",
      "type": "EventFormat",
    },
    "id": "ZSPXJBAAACIAiERm",
    "image": {
      "16:9": {
        "alt": "An laptop sitting on a desk, on the laptop screen is an etching from a 17th-century book where a woman is applying cream to her face while looking at a mirror.",
        "copyright": "Exploring Research Seminar | Photo: Kathleen Arundell. Image on screen: from the frontispiece to The Accomplished Ladies Delight, Hannah Woolley, 1675|  Wellcome Collection | | CC-BY | |",
        "dimensions": {
          "height": 1800,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": 0,
          "y": 0,
          "zoom": 1.25,
        },
        "id": "ZSamWhAAACAAlKe0",
        "url": "https://images.prismic.io/wellcomecollection/48caaf01-f243-4117-a4af-36c386acd345_EP_002514_001.jpg?auto=compress,format&rect=0,0,2560,1440&w=3200&h=1800",
      },
      "32:15": {
        "alt": "An laptop sitting on a desk, on the laptop screen is an etching from a 17th-century book where a woman is applying cream to her face while looking at a mirror.",
        "copyright": "Exploring Research Seminar | Photo: Kathleen Arundell. Image on screen: from the frontispiece to The Accomplished Ladies Delight, Hannah Woolley, 1675|  Wellcome Collection | | CC-BY | |",
        "dimensions": {
          "height": 1500,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": 0,
          "y": -150,
          "zoom": 1.25,
        },
        "id": "ZSamWhAAACAAlKe0",
        "url": "https://images.prismic.io/wellcomecollection/48caaf01-f243-4117-a4af-36c386acd345_EP_002514_001.jpg?auto=compress,format&rect=0,120,2560,1200&w=3200&h=1500",
      },
      "alt": "An laptop sitting on a desk, on the laptop screen is an etching from a 17th-century book where a woman is applying cream to her face while looking at a mirror.",
      "copyright": "Exploring Research Seminar | Photo: Kathleen Arundell. Image on screen: from the frontispiece to The Accomplished Ladies Delight, Hannah Woolley, 1675|  Wellcome Collection | | CC-BY | |",
      "dimensions": {
        "height": 1440,
        "width": 2560,
      },
      "edit": {
        "background": "#fff",
        "x": 0,
        "y": 0,
        "zoom": 1,
      },
      "id": "ZSamWhAAACAAlKe0",
      "square": {
        "alt": "An laptop sitting on a desk, on the laptop screen is an etching from a 17th-century book where a woman is applying cream to her face while looking at a mirror.",
        "copyright": "Exploring Research Seminar | Photo: Kathleen Arundell. Image on screen: from the frontispiece to The Accomplished Ladies Delight, Hannah Woolley, 1675|  Wellcome Collection | | CC-BY | |",
        "dimensions": {
          "height": 3200,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": -1244,
          "y": 0,
          "zoom": 2.2222222222222223,
        },
        "id": "ZSamWhAAACAAlKe0",
        "url": "https://images.prismic.io/wellcomecollection/48caaf01-f243-4117-a4af-36c386acd345_EP_002514_001.jpg?auto=compress,format&rect=560,0,1440,1440&w=3200&h=3200",
      },
      "type": "PrismicImage",
      "url": "https://images.prismic.io/wellcomecollection/48caaf01-f243-4117-a4af-36c386acd345_EP_002514_001.jpg?auto=compress,format",
    },
    "interpretations": [
      {
        "id": "YzGD9hEAAEAKfx0E",
        "label": "Captioned (online)",
        "type": "EventInterpretation",
      },
    ],
    "locations": [
      {
        "id": "ef04c8e3-26be-4fbc-9bef-f52589ebc56c",
        "label": "Online",
        "type": "EventLocation",
      },
    ],
    "times": [
      {
        "endDateTime": 2023-11-14T16:30:00.000Z,
        "startDateTime": 2023-11-14T15:00:00.000Z,
      },
    ],
    "title": "Recipes for Early Modern Beauty",
    "type": "Event",
  },
  "id": "ZSPXJBAAACIAiERm",
  "query": {
    "caption": "Join an online conversation about early modern health and beauty routines, and what they reveal about western European women’s lives.",
    "linkedIdentifiers": [
      "Wd-QYCcAACcAoiJS",
      "YzGD9hEAAEAKfx0E",
      "Wyu-rCoAACsACpm_",
      "ZIAyLRAAACEAANjv",
      "ZSP5ThAAACIAiMsF",
      "ZSP7IhAAACMAiNPU",
      "Wyu-0CoAACkACppp",
      "ZSP8zBAAACEAiNuV",
      "W5Zj_CYAACQALK5B",
    ],
    "series": [
      {
        "contributors": [],
        "id": "W5Zj_CYAACQALK5B",
        "title": "Exploring Research",
      },
    ],
    "title": "Recipes for Early Modern Beauty",
  },
}
`;

exports[`eventDocument transformer transforms articles from Prismic to the expected format (document: events/ZTevFxAAACQAyHEk) 1`] = `
{
  "display": {
    "format": {
      "id": "WcKmiysAACx_A8NR",
      "label": "Workshop",
      "type": "EventFormat",
    },
    "id": "ZTevFxAAACQAyHEk",
    "image": {
      "16:9": {
        "alt": "People sitting around a table with laptops editing Wikipedia pages and listening to a facilitator.",
        "copyright": "Wikithon Event | Photo: Thomas S. G. Farnetti | Wellcome Collection | | CC-BY-NC ",
        "dimensions": {
          "height": 1800,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": 0,
          "y": 0,
          "zoom": 0.920863309352518,
        },
        "id": "ZTfEHxAAACQAyNMD",
        "url": "https://images.prismic.io/wellcomecollection/89a8299e-a07e-40b6-8b02-020e29442b70_EP_000607_088+16x9+4K.jpg?auto=compress,format&rect=0,0,3475,1955&w=3200&h=1800",
      },
      "32:15": {
        "alt": "People sitting around a table with laptops editing Wikipedia pages and listening to a facilitator.",
        "copyright": "Wikithon Event | Photo: Thomas S. G. Farnetti | Wellcome Collection | | CC-BY-NC ",
        "dimensions": {
          "height": 1500,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": 0,
          "y": -150,
          "zoom": 0.920863309352518,
        },
        "id": "ZTfEHxAAACQAyNMD",
        "url": "https://images.prismic.io/wellcomecollection/89a8299e-a07e-40b6-8b02-020e29442b70_EP_000607_088+16x9+4K.jpg?auto=compress,format&rect=0,163,3475,1629&w=3200&h=1500",
      },
      "alt": "People sitting around a table with laptops editing Wikipedia pages and listening to a facilitator.",
      "copyright": "Wikithon Event | Photo: Thomas S. G. Farnetti | Wellcome Collection | | CC-BY-NC ",
      "dimensions": {
        "height": 1955,
        "width": 3475,
      },
      "edit": {
        "background": "#fff",
        "x": 0,
        "y": 0,
        "zoom": 1,
      },
      "id": "ZTfEHxAAACQAyNMD",
      "square": {
        "alt": "People sitting around a table with laptops editing Wikipedia pages and listening to a facilitator.",
        "copyright": "Wikithon Event | Photo: Thomas S. G. Farnetti | Wellcome Collection | | CC-BY-NC ",
        "dimensions": {
          "height": 3200,
          "width": 3200,
        },
        "edit": {
          "background": "#fff",
          "x": -1243,
          "y": 0,
          "zoom": 1.6368286445012787,
        },
        "id": "ZTfEHxAAACQAyNMD",
        "url": "https://images.prismic.io/wellcomecollection/89a8299e-a07e-40b6-8b02-020e29442b70_EP_000607_088+16x9+4K.jpg?auto=compress,format&rect=760,0,1955,1955&w=3200&h=3200",
      },
      "type": "PrismicImage",
      "url": "https://images.prismic.io/wellcomecollection/89a8299e-a07e-40b6-8b02-020e29442b70_EP_000607_088+16x9+4K.jpg?auto=compress,format",
    },
    "interpretations": [],
    "locations": [
      {
        "id": "XVqn7RMAACIAxvVM",
        "label": "Wellcome Kitchen",
        "type": "EventLocation",
      },
    ],
    "times": [
      {
        "endDateTime": 2023-11-22T14:00:00.000Z,
        "startDateTime": 2023-11-22T11:00:00.000Z,
      },
    ],
    "title": "Sexing Up the Internet",
    "type": "Event",
  },
  "id": "ZTevFxAAACQAyHEk",
  "query": {
    "caption": "Join Sophia and Wellcome digital editor Alice White at a Wikipedia edit-a-thon, and make your own contributions to Wikipedia pages about sexual health.",
    "linkedIdentifiers": [
      "WcKmiysAACx_A8NR",
      "XVqn7RMAACIAxvVM",
      "Wyu-0CoAACkACppp",
      "ZTev3RAAACYAyHTx",
      "WfLM2yoAAKKpVrxD",
      "Wyu-rCoAACsACpm_",
      "ZVIKWBIAACcAZMlz",
    ],
    "series": [],
    "title": "Sexing Up the Internet",
  },
}
`;
